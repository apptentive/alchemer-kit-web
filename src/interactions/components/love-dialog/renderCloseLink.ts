/**
 * Renders a given Love Dialog's set of actions.
 *
 * @param {Function} onClickHandler - The action to take when the close button is clicked.
 * @returns {HTMLDivElement} - The actions DOM Node.
 */
const renderCloseLink = (onClickHandler: (event: Event) => void) => {
  const closeLink = document.createElement('div');
  closeLink.classList.add('close-love-dialog');
  closeLink.textContent = 'âœ•';
  closeLink.addEventListener('click', onClickHandler);

  return closeLink;
};

export default renderCloseLink;
