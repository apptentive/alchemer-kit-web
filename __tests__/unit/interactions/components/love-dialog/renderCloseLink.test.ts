import renderCloseLink from '../../../../../src/interactions/components/love-dialog/renderCloseLink';

const _onClickHandler = jest.fn();

describe('renderCloseLink', () => {
  afterEach(() => {
    _onClickHandler.mockClear();
  });

  test('properly renders close link', () => {
    const output = renderCloseLink(_onClickHandler);

    expect(output).toMatchSnapshot();
  });

  test('properly attaches click handler', () => {
    document.body.append(renderCloseLink(_onClickHandler));

    const closeButton = document.querySelector<HTMLButtonElement>('.close-love-dialog')!;
    closeButton.click();

    expect(_onClickHandler).toHaveBeenCalledTimes(1);

    document.body.innerHTML = '';
  });
});
