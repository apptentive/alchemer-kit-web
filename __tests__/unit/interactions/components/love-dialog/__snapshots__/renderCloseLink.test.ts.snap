// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderCloseLink properly renders close link 1`] = `
<div
  class="close-love-dialog"
>
  âœ•
</div>
`;
